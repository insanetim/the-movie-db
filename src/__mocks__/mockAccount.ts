import { IAccount } from 'src/interfaces/account.interface'

const mockAccount: IAccount = {
  avatar: {
    gravatar: {
      hash: '/image'
    },
    tmdb: {
      avatar_path: null
    }
  },
  id: 1234,
  include_adult: true,
  iso_639_1: 'en',
  iso_3166_1: 'en',
  name: '',
  username: 'test/username'
}

export default mockAccount
