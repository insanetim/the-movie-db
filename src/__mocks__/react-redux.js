export const dispatch = jest.fn()

export const useDispatch = jest.fn(() => dispatch)
export const useSelector = jest.fn(fn => fn())
