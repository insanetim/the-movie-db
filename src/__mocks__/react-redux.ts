const dispatch = jest.fn()
const getState = jest.fn()

const useDispatch = jest.fn(() => dispatch)
const useSelector = jest.fn(fn => fn())

export { dispatch, getState, useDispatch, useSelector }
