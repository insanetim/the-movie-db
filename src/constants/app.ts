export const APP_NAME = 'THE MOVIE DB'
export const API_URL = 'https://api.themoviedb.org/3/'
export const ACCESS_TOKEN_AUTH =
  'eyJhbGciOiJIUzI1NiJ9.eyJhdWQiOiI1N2RiYjI4M2M4MDBjY2QwYmM5YTczNTJjN2Y1ZmEzYiIsInN1YiI6IjYzMmM4YThkZTNmYTJmMDA3ZmQ0ZjY5OSIsInNjb3BlcyI6WyJhcGlfcmVhZCJdLCJ2ZXJzaW9uIjoxfQ.nMVY53gL3_w8JAHI-N_uKQyXpvT9TsZ4_p8nhJSnAdQ'
export enum NOTIFICATION_TYPE {
  ERROR = 'error',
  SUCCESS = 'success'
}
export const NOTIFICATION_DURATION = 2.5
