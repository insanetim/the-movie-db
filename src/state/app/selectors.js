import { path } from 'ramda'

export const modalTypeSelector = path(['app', 'modal', 'modalType'])
export const modalPropsSelector = path(['app', 'modal', 'modalProps'])
export const notificationsSelector = path(['app', 'notifications'])
