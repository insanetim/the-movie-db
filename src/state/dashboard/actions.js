import * as types from './types'

export const fetchTrending = page => ({
  type: types.FETCH_TRENDING,
  payload: page
})

export const fetchTrendingRequest = () => ({
  type: types.FETCH_TRENDING_REQUEST
})

export const fetchTrendingSuccess = movies => ({
  type: types.FETCH_TRENDING_SUCCESS,
  payload: movies
})

export const fetchTrendingError = error => ({
  type: types.FETCH_TRENDING_ERROR,
  payload: error
})

export const setTrendingPage = page => ({
  type: types.SET_TRENDING_PAGE,
  payload: page
})

export const fetchSearch = ({ page, query }) => ({
  type: types.FETCH_SEARCH,
  payload: { page, query }
})

export const fetchSearchRequest = () => ({
  type: types.FETCH_SEARCH_REQUEST
})

export const fetchSearchSuccess = movies => ({
  type: types.FETCH_SEARCH_SUCCESS,
  payload: movies
})

export const fetchSearchError = error => ({
  type: types.FETCH_SEARCH_ERROR,
  payload: error
})

export const setSearchPage = page => ({
  type: types.SET_SEARCH_PAGE,
  payload: page
})
