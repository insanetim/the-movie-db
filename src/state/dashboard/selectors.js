import { path } from 'ramda'

export const trendingSelector = path(['dashboard', 'trending'])
export const searchSelector = path(['dashboard', 'search'])
