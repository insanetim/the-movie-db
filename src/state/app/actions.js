import { v4 as uuid } from 'uuid'

import { NOTIFICATION_DURATION, NOTIFICATION_TYPE } from 'src/constants'

import * as types from './types'

export const showModal = payload => ({
  type: types.SHOW_MODAL,
  payload
})

export const hideModal = () => ({
  type: types.HIDE_MODAL
})

export const showNotification = ({
  id = uuid(),
  messageType = NOTIFICATION_TYPE.SUCCESS,
  messageText,
  duration = NOTIFICATION_DURATION
}) => ({
  type: types.SHOW_NOTIFICATION,
  payload: {
    id,
    messageType,
    messageText,
    duration
  }
})

export const hideNotification = payload => ({
  type: types.HIDE_NOTIFICATION,
  payload
})
