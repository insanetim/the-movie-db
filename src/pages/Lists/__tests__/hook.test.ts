import { act, renderHook } from '@testing-library/react'
import { useSearchParams } from 'react-router-dom'
import mockAccount from 'src/__mocks__/mockAccount'
import { dispatch } from 'src/__mocks__/react-redux'
import { showModal } from 'src/store/app/actions'
import { fetchLists } from 'src/store/lists/actions'
import { accountSelector } from 'src/store/session/selectors'

import useContainer from '../hook'

jest.mock('src/store/lists/actions')

jest.mock('src/store/session/selectors', () => ({
  accountSelector: jest.fn(() => null)
}))

jest.mock('src/store/lists/selectors', () => ({
  listsErrorSelector: jest.fn(() => null),
  listsLoadingSelector: jest.fn(() => true),
  listsSelector: jest.fn(() => null)
}))

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useSearchParams: jest.fn()
}))
const searchParams = new URLSearchParams()
const setSearchParams = jest.fn()
jest.mocked(useSearchParams).mockReturnValue([searchParams, setSearchParams])

describe('Lists useContainer hook', () => {
  it('should match snapshot', () => {
    const { result } = renderHook(useContainer)

    expect(result.current).toMatchSnapshot()
  })

  it('should check `handlePagination` method', () => {
    const { result } = renderHook(useContainer)

    act(() => {
      result.current.handlePagination(3)
    })

    expect(setSearchParams).toHaveBeenCalledWith({ page: '3' })
  })

  it('should check `handleCreateList` method', () => {
    let onSuccess = () => {
      return
    }
    const { result } = renderHook(useContainer)

    act(() => {
      onSuccess = result.current.handleCreateList()
    })
    onSuccess()

    expect(dispatch).toHaveBeenCalledWith(
      showModal({
        modalProps: { onSuccess },
        modalType: 'MODAL_CREATE_LIST'
      })
    )
    expect(dispatch).toHaveBeenCalledWith(fetchLists('1'))
  })

  it('should check `handleCreateList` method with page !== 1', () => {
    const searchParams = new URLSearchParams({ page: '3' })
    jest
      .mocked(useSearchParams)
      .mockReturnValue([searchParams, setSearchParams])
    let onSuccess = () => {
      return
    }
    const { result } = renderHook(useContainer)

    act(() => {
      onSuccess = result.current.handleCreateList()
    })
    onSuccess()

    expect(dispatch).toHaveBeenCalledWith(
      showModal({
        modalProps: { onSuccess },
        modalType: 'MODAL_CREATE_LIST'
      })
    )
    expect(setSearchParams).toHaveBeenCalledWith({})
  })

  it('should check `useEffect` method with account', () => {
    jest.mocked(accountSelector).mockReturnValueOnce(mockAccount)
    renderHook(useContainer)

    expect(dispatch).toHaveBeenCalledWith(fetchLists('1'))
  })
})
