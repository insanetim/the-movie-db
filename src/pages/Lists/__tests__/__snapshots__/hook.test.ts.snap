// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Lists useContainer hook should match snapshot 1`] = `
{
  "error": null,
  "handleCreateList": [Function],
  "handlePagination": [Function],
  "lists": null,
  "loading": true,
}
`;
